client/docs/modules.md

```diff
@@ -0,0 +1,39 @@
+# Monorepo: Modules
+
+Modules are the packages that can be found in the `modules/` folder. They're intended for pure ESM output, and are compiled rapidly using Estrella, an esbuild-based utility. They have an important file structure (don't change it), which looks like the following:
+
+```
+module:
+- src/
+  - index.ts
+  - **/*.ts
+- tests/
+  - *.ts
+- package.json
+- tsconfig.json
+```
+
+The `src/` folder is compiled into a `dist/` folder when the module is built. Similarly, files in the `tests/` folder are compiled into the `tests/dist/` folder when the tests are built.
+
+Other than the inheritance of development dependencies and tooling configurations, these modules act like fairly normal NPM packages. If you need to add a dependency to one, you can either navigate to the module and do a normal `npm i <dep>` command, or manually add it to the `package.json`. After you do that, you need to make sure to run the `npm run bootstrap` command at the root. This ensures that all the fragile symlinks get restored by Lerna.
+
+To improve performance of the monorepo, building and watching of modules isn't done per module, but instead globally at root. It is recommended that you just build the entire monorepo rather than trying to build individual packages. This is not an insane recommendation because module building is 100% incremental. If a module has not changed, it won't get rebuilt.
+
+There is a template for making modules, and it can be found in the `templates/module-template` folder.
+
+Here is how to use that template:
+
+1. Create a folder in `modules/` that is the name of your package.
+  It isn't strictly required that it is the name of your package, but it gets confusing otherwise.
```

Comment:
This seems more like a footnote. I don't think we should ever have a case where the name and directory mismatch, the documentation should just assert that they're the same.

Reply:
Well, we have that case already when you prefix with `node-`. I feel like that information is important because I don't want to hide how the repo works fundamentally.